packages/shared/src/util/features.ts

```diff
@@ -736,6 +756,119 @@ export function getParsedPrereqCondition(condition: string) {
   return undefined;
 }
 
+export function getReviewSetting(
+  requireReviewSettings: RequireReview[],
+  feature: FeatureInterface
+): RequireReview | undefined {
+  // check projects
+  for (const reviewSetting of requireReviewSettings) {
+    // match first value found empty means all projects
+    if (
+      (feature?.project && reviewSetting.projects.includes(feature?.project)) ||
+      reviewSetting.projects.length === 0
+    ) {
+      return reviewSetting;
+    }
+  }
+}
+
+export function checkEnvironmentsMatch(
+  feature: FeatureInterface,
+  environments: string[],
+  reviewSetting: RequireReview
+) {
+  for (const env of reviewSetting.environments) {
+    if (environments.includes(env)) {
+      return true;
+    }
+  }
+  return reviewSetting.environments.length === 0;
+}
+export function featureRequiresReview(
+  feature: FeatureInterface,
+  requestedEnvironments: string[],
+  baseRevision: FeatureRevisionInterface,
+  revision: FeatureRevisionInterface,
+  settings?: OrganizationSettings
+) {
+  const requiresReviewSettings = settings?.requireReviews;
+  //legacy check
+  if (
+    requiresReviewSettings === undefined ||
+    requiresReviewSettings === true ||
+    requiresReviewSettings === false
+  ) {
+    return !!requiresReviewSettings;
+  }
+  const reviewSetting = getReviewSetting(requiresReviewSettings, feature);
+  const defaultValueChanged =
+    baseRevision.defaultValue !== revision.defaultValue;
+  if (!reviewSetting || !reviewSetting.requireReviewOn) {
+    return false;
+  }
+  if (defaultValueChanged) {
+    return true;
+  }
+  return checkEnvironmentsMatch(feature, requestedEnvironments, reviewSetting);
+}
+
+export function resetReviewOnChange(
+  feature: FeatureInterface,
+  requestedEnvironments: string[],
+  defaultValueChanged: boolean,
+  settings?: OrganizationSettings
```

Comment:
Change these to named arguments.  This function is called all over the place and I keep having to check back here to see what the arguments mean.